src/cargo/ops/vendor.rs

```diff
@@ -35,7 +40,7 @@ pub fn vendor(ws: &Workspace<'_>, opts: &VendorOptions<'_>) -> CargoResult<()> {
         extra_workspaces.push(ws);
     }
     let workspaces = extra_workspaces.iter().chain(Some(ws)).collect::<Vec<_>>();
-    let _lock = gctx.acquire_package_cache_lock(CacheLockMode::MutateExclusive)?;
+    let _lock = gctx.acquire_package_cache_lock(CacheLockMode::DownloadExclusive)?;
```

Comment:
Technically, this change belongs in 9fb4dbcb9da1e85278960474f5f6a914e11ec501 ("fix(vendor): remove workaround") since this commit is still making edits to the cache

Reply:
Nice catch. Updated.